### PR TITLE
Fix 299

### PR DESCRIPTION
**Pull request by steffenheyne**
this makes the handling of provided gtf files with annotation more generic,
fix some other issues like that the transcript fasta file still contained introns although -split was used
fix sleuth not working correctly and some minor things

### CODE REVIEW

**Comment by steffenheyne** [comment]
scRNA-seq and RNA-seq run through without error for me now

File: snakePipes/shared/rules/filter_annotation.snakefile [hunk 1]
```diff
@@ -1,54 +1,49 @@
...
+        """ pos=match($0,"transcript_name.([^[:space:]]+)",a); if (pos!=0) tna=a[1]; else tna=tid; """
+        """ pos=match($0,"transcript_[bio]*type.([^[:space:]]+)",a); if (pos!=0) tt=a[1]; else tt="NA"; """
+        """ pos=match($0,"gene_name.([^[:space:]]+)",a);  if (pos!=0) gna=a[1]; else gna=gid; """
+        """ pos=match($0,"gene_[bio]*type.([^[:space:]]+)",a); if (pos!=0) gt=a[1]; else gt="NA"; """
+        """ pos=match($0,"(transcript_support_level.[^[:space:]]+)",a); if (pos!=0) tsl=a[1]; else tsl="transcript_support_level NA"; """
+        """ pos=match($0,"[[:space:]](level.[^[:space:]]+)",a); if (pos!=0) lvl=a[1] ; else lvl="level NA"; """
+        """ pos=match($0,"tag.basic"); if (lvl!~"NA"){{if (pos==0) basic="full"; else basic="basic"}} else basic="NA"; """
+        """ OFS="\\t"; print tid,tna,tt,gid,gna,gt,"gencode",basic,"transcript_support_level",tsl,"level",lvl}}' | """
+        """ sort | uniq | sort -k1,1) | """
+        """ awk '{{OFS="\\t";print $2,$3,$4,$1,$5,$6,$7,$8,$9,$10,$11,$12,$1,$13,$14,$15,$16,$17,$18,$19,$20,$21,$22,$23}}' > {output.bed_annot} """
```
Comment (dpryan79) [inline snakePipes/shared/rules/filter_annotation.snakefile]: I'd still prefer this to be a more readable python script long-term, but that can be saved for another day.

File: snakePipes/workflows/RNA-seq/Snakefile [hunk 2]
```diff
@@ -245,7 +245,7 @@ rule all:
         run_Trimming(trim, fastqc),
         "Annotation/genes.annotated.bed",
         "Annotation/genes.filtered.bed",
-#        "Annotation/genes.filtered.saf",    # this file is used for featureCounts later on!
+        "Annotation/genes.filtered.gtf",
```
Comment (dpryan79) [inline snakePipes/workflows/RNA-seq/Snakefile]: Thank you!

**Comment by steffenheyne** [comment]
thanks!

File: snakePipes/shared/rules/filter_annotation.snakefile (hunk 1 above)
Comment (steffenheyne) [reply snakePipes/shared/rules/filter_annotation.snakefile]: yeah we can do this at some point, I tried to simplify the pipe as much as possible right now

**Pull request merged**